src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        result[i] = { ...Object.assign(currentState, actions[i].extraData) };
```

Comment:
Using both spread operator and Object.assign is redundant here. You should use either one to create a copy and add properties. For example: `const newState = { ...currentState, ...actions[i].extraData };`